pkg/kubelet/certificate/kubelet.go

```diff
@@ -70,9 +72,32 @@ func newGetTemplateFn(nodeName types.NodeName, getAddresses func() []v1.NodeAddr
 	}
 }
 
+// getNodeAddressesFromInformer returns the node addresses from the informer cache.
+// On external cloud providers nodes it waits for the external cloud provider to
+// remove the taint, since the node may start with some initial addresses that are changed
+// later by the cloud provider and we need to use only the definitive addresses for the
+// certificate request.
+func getNodeAddressesFromInformer(nodeName types.NodeName, nodeLister corelisters.NodeLister) []v1.NodeAddress {
+	node, err := nodeLister.Get(string(nodeName))
+	if err != nil {
+		klog.ErrorS(err, "fail to obtain node from local cache", "node", nodeName)
+		return nil
+	}
+	if len(node.Status.Addresses) == 0 {
```

Comment:
This is my lack of experience with the Kubernetes codebase, is it possible that `node.Status` is nil at this stage?

Reply:
Looking at other parts of the code base, it seems to be safe to assume that `node.Status` isn't nil

Reply:
if no node is found we get an err NotFound IIRC, so at this point we should have a node object

Reply:
The part I was wondering about, is then a node is returned but doesn't yet contain status, that way `node.Status` would be nil, and `node.Status.Addresses` would return a nil pointer exception.

But I see that this is safe (I think I was confused with pointer references), since this can't happen.